app/javascript/packages/document-capture/components/full-screen.jsx

```diff
@@ -34,6 +34,11 @@ function FullScreen({ onRequestClose = () => {}, children }) {
     return trapRef.current.deactivate;
   }, []);
 
+  useEffect(() => {
+    document.body.classList.add('has-full-screen-overlay');
+    return () => document.body.classList.remove('has-full-screen-overlay');
```

Comment:
Would we ever have two full-screen components that stack on top of each other? Is there a way to scope this? Or should we add a global counter or something? Like something along the lines of:

```js
// somewhere at like a static scope
let numFullScreen = 0;

  useEffect(() => {
    ++numFullScreen;
    document.body.classList.add('has-full-screen-overlay');
    return () => {
      if (--numFullScreen == 0) {
        document.body.classList.remove('has-full-screen-overlay');
      }
    }
```

Reply:
> Would we ever have two full-screen components that stack on top of each other? Is there a way to scope this? Or should we add a global counter or something?

That's a good observation! From a usability perspective, we _shouldn't want_ there ever to be two full screen components active at the same time. But technically speaking, you'd be right that it's possible, and that the unmounting of one might inadvertently remove the body class still required by the other.

Your suggestion makes sense to me. I'll incorporate it!

Reply:
Updated in 15abbfa.

Is it also fair for me to take your suggested snippet as tacit agreement to disable [`no-plusplus`](https://eslint.org/docs/rules/no-plusplus)? 😉  I've done so (with justification) in 58825ff.

Reply:
ohhh noooo what have I unleashedddddd 😜 I just used the `++` out of habit, didn't realize we'd need to change a lint for it

1. I left out the conditional on adding, because `classList` deduplicate, so adding a second time is a no-op
2. I also intentionally used prefix, not postfix because I think mental math is easier (and it translates to `+=` notation clearer if we wanted to bring that lint back). I think that `if (x++ == 0)` is really confusing to read

That said, none of this is that important to me that I need a PR to fix it... just wanted to clear up what I meant

Reply:
> 1. I left out the conditional on adding, because `classList` deduplicates for us, so adding a second time is a no-op

Yeah, I'd paused to think about that, and admittedly couldn't recall if this was one of those cases†, and came to the meta-conclusion that the potential confusion itself might warrant some extra clarification, even if not strictly necessary. 

† Also failing to recall off the top of my head, but I know there's another of these `add`-like set implementations which _doesn't_ behave this way.

> 2\. I also intentionally used prefix, not postfix because I think mental math is easier (and it translates to `+=` notation clearer if we wanted to bring that lint back). I think that `if (x++ == 0)` is really confusing to read

Ah, I think what happened here is that I had initially considered a more "clever" implementation:

```js
if (!activeInstances++) {
  document.body.classList.add('has-full-screen-overlay');
}

return () => --activeInstances || document.body.classList.remove('has-full-screen-overlay');
```

I eventually opted toward something more readable, but apparently forgot that the order flipping became unnecessary.

> I just used the `++` out of habit, didn't realize we'd need to change a lint for it

I'd be fine to back it out. I'd encountered it once before, where [this implementation](https://github.com/18F/identity-idp/blob/85418e8f2fdcde51446425a48ec526a898241354/app/javascript/packages/document-capture/hooks/use-instance-id.js#L12-L15) could have been simplified:

```js
return useMemo(() => String(instances++), []);
```

I'd also chatted with @ironman5366 at the time, who'd mentioned similar frustrations about the rule.

That being said, I can see arguments in favor of it being unnecessarily unclear, where even if the `+=` is not nearly as concise, it'd be easier to read.

Reply:
> > 1. I left out the conditional on adding, because `classList` deduplicates for us, so adding a second time is a no-op
> 
> Yeah, I'd paused to think about that, and admittedly couldn't recall if this was one of those cases†, and came to the meta-conclusion that the potential confusion itself might warrant some extra clarification, even if not strictly necessary.

Plus a secondary concern that touching DOM APIs can often be pretty non-performant, and better to rely on logic which can be known to be fast.

Possibly related: https://gist.github.com/paulirish/5d52fb081b3570c81e3a#gistcomment-2351596

Reply:
> † Also failing to recall off the top of my head, but I know there's another of these `add`-like set implementations which _doesn't_ behave this way.

I believe what I had mind was [`URLSearchParams.append`](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/append).